### ISSUE TITLE
crash in sendTransformed : this was nullptr

### ISSUE THREAD

**Issue by ad34** [uNetworking/uWebSockets#733]
* What operating system are you running?

  * Windows
* Are you running any CPU architechute out of the normal?
no
* What platform/language are you using?
  * C++

* Are you using SSL, if so, how?
  * SSL via ALB (AWS)
* What version of uws are you using?
  * master
* What features of µWS are involved in the issue?

  * Client support (WebSocket)

* How does the issue reproduce?
  * Randomly (1%)

* Can you show me some short code that triggers your issue or any other important information?

Sometimes when I send a message I get a crash in sendTransformed (hasEmptyQueue)

here is how the callstack looks like, I changed some functions names and remove the begining of the callstach but it gives an idea of the context

MyExe.exe!uS::Socket::sendTransformed<`uWS::WebSocket<0>::send'::`2'::WebSocketTransformer,`uWS::WebSocket<0>::send'::`2'::TransformData>(const char * message, unsigned __int64 length, void(*)(void *, void *, bool, void *) callback, void * callbackData, uWS::WebSocket<0>::send::__l2::TransformData transformData) Line 363	C++
 	MyExe.exe!MyManager::SendMyResult(RESOURCE_LUA * _lua, int _index) Line 963	C++
 	MyExe.exe!MyLuaBinding(lua_State * L) Line 37	C++
 	MyExe.exe!lj_BC_FUNCC()	Unknown
 	MyExe.exe!lua_pcall(lua_State * L, int nargs, int nresults, int errfunc) Line 1130	C
 	MyExe.exe!lua_XCall(RESOURCE_LUA * resource, int nargs, int nresults) Line 587	C++
 	MyExe.exe!LUA_EVALSTACK(PUSHPOP_Transmitter & Trans, RESOURCE_LUA * ResourceSrc, Pusher & Result, unsigned int SubTableMode) Line 251	C++
 	MyExe.exe!`anonymous namespace'::CallEvent(const `anonymous-namespace'::EventContext & Context) Line 135	C++
 	MyExe.exe!`anonymous namespace'::EventChannel::Wait() Line 318	C++
 	MyExe.exe!LuaEventTriggerWaitFinish(lua_State * L) Line 512	C++
 	MyExe.exe!lj_BC_FUNCC()	Unknown
 	MyExe.exe!lua_pcall(lua_State * L, int nargs, int nresults, int errfunc) Line 1130	C

Basically uws is this project is a WebSocket client, we use a LUA virtual machine .
A lua function that binds a C++ function on the main thread is called to send a message. the message is converted to MsgPack and sent via a websocket 

the message size is 0x61b and the format is binary.

A exception is throw because this pointer does not exists anymore

Unhandled exception thrown: read access violation.
this was nullptr.
If there is a handler for this exception, the program may be safely continued.

The repo rate is very low (around 1%)

**Comment by alexhultman** [comment]
This is a too complex case to report. This is most likely an issue on your end. In any case it is impossible for me to debug this and I won't do so.

**Comment by alexhultman** [comment]
You need to destill the issue down to a very minimal case without any Lua VM involved, etc.

**Comment by ad34** [comment]
I had the same callstack with a call made from C++ side last week . I will post it when I reproduce the issue.
